engine/base/runtime/src/main/scala/pl/touk/nussknacker/engine/baseengine/ScenarioInterpreterFactory.scala

```diff
@@ -171,7 +171,8 @@ object ScenarioInterpreterFactory {
     private def compileWithCompilationErrors(node: SplittedNode[_], validationContext: ValidationContext): ValidatedNel[ProcessCompilationError, Node] =
       processCompilerData.subPartCompiler.compile(node, validationContext)(compiledProcess.metaData).result
 
-    private def customComponentContext(nodeId: String) = CustomComponentContext[F](nodeId, lazyParameterInterpreter, capabilityTransformer)
+    private def customComponentContext(nodeId: String) = CustomComponentContext[F](nodeId, lazyParameterInterpreter, capabilityTransformer,
```

Comment:
:/ let's add at least a TODO to remove this duplication of creating IdGenerator

Reply:
added